src/dmd/dmangle.d

```diff
@@ -662,6 +666,24 @@ public:
                 buf.printf("%u__S%u", ndigits + 3, localNum);
             }
         }
+        else // No parent means it's the root symbol
+        {
+            const(char)[] symbolOrigin = leaf.loc.filename ? leaf.loc.canonicalFilename :
+                                         leaf.parent && leaf.parent.loc.filename ? leaf.parent.loc.canonicalFilename : null;
+                                         // Auto generated struct members have no defined source origin (aka Loc)
+                                         // so we use their parent (the struct itself) for extracting the source file path
+            // Use filename of leaf node to compare against -mangle-prefix paths
+            foreach_reverse (mangleSuffix; global.params.manglePrefixList)
+            {
+                if (mangleSuffix.type == FileType.file && FileName.equals(mangleSuffix.path, symbolOrigin) ||
```

Comment:
This if condition is a bit ugly, I suggest breaking it up (i.e. modern compilers let us give the clauses their own variables at no additional cost)